src/_cffi_src/openssl/evp.py

```diff
@@ -180,6 +180,11 @@
     EVP_MD_CTX_free(md);
 }
 
+#if !CRYPTOGRAPHY_IS_LIBRESSL && !CRYPTOGRAPHY_IS_BORINGSSL && \
+        !CRYPTOGRAPHY_OPENSSL_300_OR_GREATER
+int EVP_PKEY_set_alias_type(EVP_PKEY *, int);
```

Comment:
This needs to be conditionally removed, as you can see right below with scrypt -- you'll also need to edit `_conditional.py`.

Reply:
> This needs to be conditionally removed, as you can see right below with scrypt -- you'll also need to edit `_conditional.py`.

Thanks for you review. `EVP_PKEY_set_alias_type` is a function only for OpenSSL 1.1.1, And there are failing compilation in openssl 3 as well as boring and libre . So we need to add some conditions to judge the type of openssl. But after adding the conditional judgment，it can't find the corresponding function symbol(`EVP_PKEY_set_alias_type`) in `_openssl.abi3.so` with OpenSSL 1.1.1. Could you give me some advice? Thanks a lot.

Reply:
Our bindings require that you put the declaration in the `FUNCTIONS` section. `CUSTOMIZATIONS` is used to handle cases where the declaration may not always work.

You should look at how we do this with `Cryptography_HAS_SCRYPT`. We add `EVP_PBE_scrypt` to `FUNCTIONS`, then create `Cryptography_HAS_SCRYPT` and set its value based on whether the function is available or not (along with a null func ptr declaration if it is not), and then we have a section in `_conditional.py` that removes the symbol from the Python object if it is unavailable.

Reply:
> Our bindings require that you put the declaration in the `FUNCTIONS` section. `CUSTOMIZATIONS` is used to handle cases where the declaration may not always work.
> 
> You should look at how we do this with `Cryptography_HAS_SCRYPT`. We add `EVP_PBE_scrypt` to `FUNCTIONS`, then create `Cryptography_HAS_SCRYPT` and set its value based on whether the function is available or not (along with a null func ptr declaration if it is not), and then we have a section in `_conditional.py` that removes the symbol from the Python object if it is unavailable.

Thank you very much for your help @reaperhulk . Now I have modified the codes according to your suggestion, and I can see the function symbol `EVP_PKEY_set_alias_type` in the `_openssl.abi3.so' with `openssl 1.1.1`. Besides, CI/CD shows that it doesn't affect other versions of `openssl`.

![a08236ef69a37cb2362c2384904a58b4](https://user-images.githubusercontent.com/6824093/209754930-87689bff-96de-4699-b63d-4b6ae625362f.png)

In addition, I will submit the `pyopenssl` PR mentioned above to `pyopenssl` as soon as possible in the past two days, and then I will let you know, Thanks again @reaperhulk .